test_conformance/basic/main.cpp

```diff
@@ -25,137 +25,137 @@
+        ADD_TEST(imagerandomcopy),
+        ADD_TEST(arrayimagecopy),
+        ADD_TEST(arrayimagecopy3d),
+        ADD_TEST(imagenpot),
+
+        ADD_TEST(vload_global),
+        ADD_TEST(vload_local),
+        ADD_TEST(vload_constant),
+        ADD_TEST(vload_private),
+        ADD_TEST(vstore_global),
+        ADD_TEST(vstore_local),
+        ADD_TEST(vstore_private),
+
+        ADD_TEST(createkernelsinprogram),
+        ADD_TEST(imagedim_pow2),
+        ADD_TEST(imagedim_non_pow2),
+        ADD_TEST(image_param),
+        ADD_TEST(image_multipass_integer_coord),
+        ADD_TEST(image_multipass_float_coord),
+        ADD_TEST(explicit_s2v_char),
+        ADD_TEST(explicit_s2v_uchar),
+        ADD_TEST(explicit_s2v_short),
+        ADD_TEST(explicit_s2v_ushort),
+        ADD_TEST(explicit_s2v_int),
+        ADD_TEST(explicit_s2v_uint),
+        ADD_TEST(explicit_s2v_long),
+        ADD_TEST(explicit_s2v_ulong),
+        ADD_TEST(explicit_s2v_float),
+        ADD_TEST(explicit_s2v_double),
+
+        ADD_TEST(enqueue_map_buffer),
+        ADD_TEST(enqueue_map_image),
+
+        ADD_TEST(work_item_functions),
+
+        ADD_TEST(astype),
+
+        ADD_TEST(async_copy_global_to_local),
+        ADD_TEST(async_copy_local_to_global),
+        ADD_TEST(async_strided_copy_global_to_local),
+        ADD_TEST(async_strided_copy_local_to_global),
+        ADD_TEST(prefetch),
+
+        ADD_TEST(kernel_call_kernel_function),
+        ADD_TEST(host_numeric_constants),
+        ADD_TEST(kernel_numeric_constants),
+        ADD_TEST(kernel_limit_constants),
+        ADD_TEST(kernel_preprocessor_macros),
+
+        ADD_TEST(parameter_types),
+        ADD_TEST(vector_creation),
+        ADD_TEST(vec_type_hint),
+        ADD_TEST(kernel_memory_alignment_local),
+        ADD_TEST(kernel_memory_alignment_global),
+        ADD_TEST(kernel_memory_alignment_constant),
+        ADD_TEST(kernel_memory_alignment_private),
+
+        ADD_TEST_FEATURE(progvar_prog_scope_misc, F_program_scope_variables),
+        ADD_TEST_FEATURE(progvar_prog_scope_uninit, F_program_scope_variables),
+        ADD_TEST_FEATURE(progvar_prog_scope_init, F_program_scope_variables),
+        ADD_TEST_FEATURE(progvar_func_scope, F_program_scope_variables),
+
+        ADD_TEST(global_work_offsets),
+        ADD_TEST(get_global_offset),
+
+        ADD_TEST_FEATURE(global_linear_id, F_version_ge(2, 0)),
+        ADD_TEST_FEATURE(local_linear_id, F_version_ge(2, 0)),
+        ADD_TEST_FEATURE(enqueued_local_size, F_version_ge(2, 0)),
+
+        ADD_TEST(simple_read_image_pitch),
+        ADD_TEST(simple_write_image_pitch),
 
 #if defined( __APPLE__ )
-    ADD_TEST( queue_priority ),
+        ADD_TEST(queue_priority),
 #endif
 
-    ADD_TEST_VERSION( get_linear_ids, Version(2, 0) ),
-    ADD_TEST_VERSION( rw_image_access_qualifier, Version(2, 0) ),
-};
+        ADD_TEST_FEATURE(get_linear_ids, F_version_ge(2, 0)),
+        ADD_TEST_FEATURE(rw_image_access_qualifier, F_version_ge(2, 0)),
```

Comment:
Since read-write images are optional in OpenCL-3.0, should we have a feature to check that clGetDeviceInfo returns a non-zero value for CL_​DEVICE_​MAX_​READ_​WRITE_​IMAGE_​ARGS?

Reply:
Yup, I was just focusing on migrating the initial checks and open PRs with this first change but this is simple enough to be done here.

Reply:
Thank you. I can confirm that with the latest changes #789 is now fixed. `test_rw_image_access_qualifier` is also successfully skipped.